File: Tyche.iOS/DataPool/Sources/DataPool/Application/GetPoolGamblerScoreUseCase.swift
import Core

public class GetPoolGamblerScoreUseCase {
    private let poolGamblerScoreRepository: PoolGamblerScoreRepository
    
    public init(poolGamblerScoreRepository: PoolGamblerScoreRepository) {
        self.poolGamblerScoreRepository = poolGamblerScoreRepository
    }
    
    public func execute(
        poolId: String,
        gamblerId: String
    ) async -> Result<PoolGamblerScore, Error> {
        return await poolGamblerScoreRepository.getPoolGamblerScore(
            poolId: poolId,
            gamblerId: gamblerId
        )
    }
}

File: Android/data/pool/src/main/java/com/felipearpa/tyche/data/pool/application/GetPoolGamblerScoreUseCase.kt
package com.felipearpa.tyche.data.pool.application

import com.felipearpa.tyche.data.pool.domain.PoolGamblerScoreRepository

class GetPoolGamblerScoreUseCase(private val poolGamblerScoreRepository: PoolGamblerScoreRepository) {
    suspend fun execute(poolId: String, gamblerId: String) =
        poolGamblerScoreRepository.getPoolGamblerScore(poolId = poolId, gamblerId = gamblerId)
}
